rcl_yaml_param_parser/src/parser.c

```diff
@@ -63,9 +65,12 @@ typedef struct namespace_tracker_s
   uint32_t num_node_ns;
   char * parameter_ns;
   uint32_t num_parameter_ns;
+  char * descriptor_key_ns;
+  uint32_t num_descriptor_key_ns;
 } namespace_tracker_t;
 
 #define PARAMS_KEY "ros__parameters"
+#define PARAMS_DESCRIPTORS_KEY "parameter__descriptors"
```

Comment:
@bpwilcox meta: why not `ros__parameter_descriptors` for consistency?

Reply:
Given it is already scoped under ros__parameters level it seemed redundant, but I'm not opposed.

This does bring up my original approach to have the `parameter__descriptors` be at the same level as `ros__parameters` (e.g. no longer scoped), so the two would be uncoupled. It seemed like much more work at the time with not too significant benefits, but I'm open to discussing it.